spansy/src/http/span.rs

```
@@ -222,12 +222,23 @@ fn request_body_len(request: &Request) -> Result<usize, ParseError> {
     // the Transfer-Encoding overrides the Content-Length
     if request
         .headers_with_name("Transfer-Encoding")
-        .next()
-        .is_some()
+        .any(|h| {
+            std::str::from_utf8(h.value.0.as_bytes())
+                .unwrap_or("")
+                .split(',')
+                .any(|v| v.trim() != "identity")
+        })
     {
-        Err(ParseError(
-            "Transfer-Encoding not supported yet".to_string(),
-        ))
+        let bad_values = request
+            .headers_with_name("Transfer-Encoding")
+            .map(|h| std::str::from_utf8(h.value.0.as_bytes()).unwrap_or(""))
```

`unwrap_or("")` will be misleading if the header value is not value utf8. Better to return a parse error indicating that.

---

Instead of returning a parse error, which might be more confusing in this case, I've updated it to use the string `{invalid utf-8}` in the error message, to signify that the header value cannot be parsed as utf-8. I'm of the mind that it's clearer that way.